net/src/icmp4/mod.rs

```diff
@@ -12,11 +17,15 @@ use std::num::NonZero;
 pub struct Icmp4(Icmpv4Header);
 
 impl Parse for Icmp4 {
-    type Error = LengthError;
+    type Error = UnreachableError;
```

Comment:
Nothing major, but these changes should be in the commit where you introduce the unreachable issues, not in the usize/u16 swap commit.